src/game/traps/lava.c

```diff
@@ -75,9 +75,25 @@ void LavaBurn(ITEM_INFO *item)
     } while (!(type & END_BIT));
 
     if (!on_lava) {
+        return false;
+    } else {
+        LavaBurn(item);
+        return true;
```

Comment:
This still produces side effects.

What I meant is doing this:

```diff
-TestLavaFloor(g_LaraItem);
+if (TestLavaFloor(g_LaraItem) {
+    LavaBurn(g_LaraItem);
+}
```

Reply:
Ah gotcha.